task.js

```diff
@@ -0,0 +1,248 @@
+// Клас який описує людину
+class Human {
+    constructor(object){
+      this.name = object.name;
+      this.surname = object.surname;
+      this.age = object.age;
+    }
+    getFullName(){
+      return `${this.name} ${this.surname}`;
+    }
+    setFullName(FullName) {
+      [this.name, this.surname] = FullName.split(' ');
+    }
+  }
+  
+  // -------------------------------------Student------------------------------
+  // Клас який успадковує клас Human
+  class Student extends Human{
+    constructor(info) {
+      super(info);
+      this.marks = info.marks;
+    }
+  
+    // повертає середню оцінку
+    getAverageMark() {
+      return this.marks.reduce((a, b) => (a + b)) / this.marks.length;
+    }
+  
+    // повертає найменшу оцінку
+    getMinMark() {
+      return Math.min(...this.marks);
+    }
+  
+    // повертає максимальну оцінку
+    getMaxMark() {
+      return Math.max(...this.marks);
+    }
+  }
+  
+  // ------------------------------------Teacher-------------------------------
+  
+  class Teacher {
+    constructor(students) {
+      this.students =  students;
+  
+      //Створює об'єкт  де ключи це Імя, а значення це масив з обєктів
+      // this.students = students.reduce((group, arr) => {
+      //   const { name } = arr;
+      //   group[name] = group[name] ?? [];
+      //   group[name].push(arr);
+      //   return group;
+      // }, {}); 
+    }
+    getstudents(){
+      console.table( this.students );
+    }
+  
+    // віддає масив імен студентів відсортованих за оцінками
+    getListOfNamesByAverageMark() {
+      let all_average_mark = [];
+      let name_all_average_mark = [];
+  
+      for (let i_name = 0; i_name < this.students.length; i_name++) {   
+        all_average_mark.push({ name : this.students[i_name].name, average_mark : this.students[i_name].getAverageMark()});
+      }
```

Comment:
Why do you need this code ? You can use `sort` only and then return names using `map`

Skeleton example:
`return [...this.group].sort(...).map(...)`